code/modules/mob/camera/eye/hologram_eye.dm

```diff
@@ -0,0 +1,57 @@
+/mob/camera/eye/hologram
+	name = "Inactive Hologram Eye"
+	ai_detector_visible = FALSE
+	acceleration = FALSE
+	relay_speech = TRUE
+	var/obj/machinery/hologram/holopad/holopad = null
+
+/mob/camera/eye/hologram/Initialize(mapload, name, origin, mob/living/user)
+	..()
+	holopad = origin
+	setLoc(holopad, 0, TRUE)
+
+/mob/camera/eye/hologram/rename_camera(new_name)
+	name = "Hologram ([new_name])"
+
+// Hologram eye relaymove code is sinfully and shamelessly ripped
+// directly from /client/Move to mimic normal run speed
+/mob/camera/eye/hologram/relaymove(mob/user, direct)
+	var/turf/step = get_turf(get_step(src, direct))
+	var/old_move_delay = user.client.move_delay
+	user.client.move_delay = world.time + world.tick_lag
+	var/delay = GLOB.configuration.movement.base_run_speed
+	if(old_move_delay + world.tick_lag > world.time)
+		user.client.move_delay = old_move_delay
+	else
+		user.client.move_delay = world.time
+	if(!(direct & (direct - 1)))
+		setLoc(step, delay)
+	else
+		var/diag_delay = delay * SQRT_2
+		setLoc(step, diag_delay)
+		if(loc == step)
+			delay = diag_delay
+	user.last_movement = world.time
+	delay = TICKS2DS(-round(-(DS2TICKS(delay))))
+	user.client.move_delay += delay
+
+/mob/camera/eye/hologram/validate_active_cameranet()
+	..(TRUE)
+
+/mob/camera/eye/hologram/setLoc(T, delay, initial = 0)
```

Comment:
lord this proc

Reply:
Is this in reference to the `validate_active_cameranet` overloading, or the ripped `relaymove` code?

If the former, I understand - it made sense to me while in the weeds of this to try to get rid of as many variable-based checks as possible, due to their inconsistent use. I felt that using procs and overloading them down the hierarchy was less fragile. I can understand how it's unclear at a glance though, and am happy to change it back to var-based checks.

If the latter, my defense is that "it works", and attempting to DRY it up would involve touching `mob_movement.dm` code that doesn't directly concern the AI, which felt a bit like scope creep for something that appears in the code only twice and in very different contexts.

Reply:
I've looked at this with fresh eyes and have refactored it into something more sane.